site/zenodo_rdm/migrator/transform/entries/records/metadata.py

```diff
@@ -140,6 +143,18 @@ def _subjects(cls, keywords):
 
         return ret
 
+    @classmethod
+    def _validate_identifier(cls, identifier):
+        """Validates the information of an identifier and completes it if needed."""
+        value = identifier.get("identifier", "")
+        scheme = identifier.get("scheme")
+        if not scheme:
+            guess = detect_identifier_schemes(value)
+            if guess:
+                identifier["scheme"] = guess[0]  # default to first guess
+            else:
+                raise InvalidIdentifier(identifier)
```

Comment:
This means the full record won't be migrated until we fix it. Leads me to question:
- For now we want to raise, so we can investigate and fix.
- Later we might want to let records pass and ignore this malformed values (i.e. remove), which might be the same action we do to fix them.

WDYT?

Reply:
Agree, the best would be just to log these errors and either skip these identifiers and get the record in or include them and deal with it later... I think the bigger task here is to figure out a way to manage logging in a better way.